File: Exercism/javascript/isbn-verifier/isbn-verifier.js
'use strict'

export class ISBN {
  constructor (code) {
    this.isbn = code.replace(/-/g, '')
  }

  isValid () {
    return /^[0-9]{9}[0-9X]$/.test(this.isbn) &&
      this.isbn
        .split('')
        .map((nr) => nr === 'X' ? 10 : Number(nr))
        .map((nr, i) => nr * (10 - i))
        .reduce((a, b) => a + b) % 11 === 0
  }
}

File: Exercism/python/isbn-verifier/isbn_verifier.py
import re


def is_valid(isbn: str) -> bool:
    """ISBN Verifier"""
    isbn = isbn.replace('-', '')
    if re.match(r'^\d{9}[0-9X]$', isbn) is None:
        return False
    return sum(int(n) * (i + 1) if n != 'X' else 10 * (i + 1)
               for i, n in enumerate(isbn[::-1])) % 11 == 0
